ocaml/libs/tracing/tracing.ml

```diff
@@ -576,17 +576,70 @@ module Export = struct
 
       let set_trace_log_dir dir = trace_log_dir := dir
 
-      let export ~trace_id ~span_json ~path : (string, exn) result =
-        try
+      module FileStream = struct
+        let max_file_size = ref 1048576
+
+        let set_max_file_size size = max_file_size := size
+
+        let file_name = ref None
+
+        let file_stream = ref None
+
+        let lock = Mutex.create ()
+
+        let new_file_name path =
           let date = Ptime_clock.now () |> Ptime.to_rfc3339 ~frac_s:6 in
-          let file =
-            path
-            ^ String.concat "-" [trace_id; "xapi"; !host_id; date]
+          let ( // ) = Filename.concat in
+          let name =
+            (path // String.concat "-" [get_service_name (); !host_id; date])
             ^ ".json"
```

Comment:
This can be part of the `String.concat`

Reply:
Sorry, I was wrong; because `-` is used to concatenate. Otherwise
```
path // String.concat "" [get_service_name (); "-"; !host_id; "-"; date; ".json"]
```

Reply:
In that case I may as well just used `^`

Reply:
[c783311](https://github.com/xapi-project/xen-api/pull/5038/commits/c783311f36d5b9e362947efad4376a27b8fd74b1)

Reply:
"^" has quadratic cost, please don't use it to join so many strings